src/routes/safe/components/Balances/SendModal/screens/ContractInteraction/RenderInputParams/index.tsx

```diff
@@ -6,36 +6,74 @@ import TextField from 'src/components/forms/TextField'
 import { composeValidators, mustBeEthereumAddress, required } from 'src/components/forms/validator'
 import Col from 'src/components/layout/Col'
 import Row from 'src/components/layout/Row'
+import Checkbox from 'src/components/forms/Checkbox'
 
 const RenderInputParams = () => {
   const {
     meta: { valid: validABI },
-  } = useField('abi', { valid: true } as any)
+  } = useField('abi', { valid: true } as Record<string, boolean>)
   const {
     input: { value: method },
   }: any = useField('selectedMethod', { value: true })
   const renderInputs = validABI && !!method && method.inputs.length
 
+  const renderInputComponent = (type, name, key, placeholder) => {
```

Comment:
Not so sure how if it'll affect performance, but it'll make this easy to test if you create a separate component with this function. WDT?